System.Device.I2c/I2cConnectionSettings.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Device.I2c
+{
+    /// <summary>
+    /// The connection settings of a device on an I2C bus.
+    /// </summary>
+    public sealed class I2cConnectionSettings
+    {
+
+        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+        private int _deviceAddress;
+
+        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+        private I2cBusSpeed _busSpeed;
+
+        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+        private I2cSharingMode _sharingMode;
+
+        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+        private int _busId;
+
+        private I2cConnectionSettings()
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="I2cConnectionSettings"/> class.
+        /// </summary>
+        /// <param name="busId">The bus ID the I2C device is connected to.</param>
+        /// <param name="deviceAddress">The bus address of the I2C device.</param>
+        public I2cConnectionSettings(int busId, int deviceAddress) : this(busId, deviceAddress, I2cBusSpeed.FastMode, I2cSharingMode.Shared)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="I2cConnectionSettings"/> class.
+        /// </summary>
+        /// <param name="busId">The bus ID the I2C device is connected to.</param>
+        /// <param name="deviceAddress">The bus address of the I2C device.</param>
+        /// <param name="busSpeed">The bus speed of the I2C device.</param>
+        /// <param name="sharingMode">The sharing mode of the I2C device.</param>
+        public I2cConnectionSettings(int busId, int deviceAddress, I2cBusSpeed busSpeed, I2cSharingMode sharingMode)
+        {
+            _busId = busId;
+            _deviceAddress = deviceAddress;
+            _busSpeed = busSpeed;
+            _sharingMode = sharingMode;
+        }
+
+        internal I2cConnectionSettings(I2cConnectionSettings other)
+        {
+            _busId = other.BusId;
+            _deviceAddress = other.DeviceAddress;
+            _busSpeed = other.BusSpeed;
+            _sharingMode = other.SharingMode;
+        }
+
+        /// <summary>
+        /// The bus ID the I2C device is connected to.
+        /// </summary>
+        public int BusId { get => _busId; }
+
+        /// <summary>
+        /// The bus address of the I2C device.
+        /// </summary>
+        public int DeviceAddress { get => _deviceAddress; }
```

Comment:
I think this could be just `{ get; }` and you could remove the field

Reply:
If the backing field is used in native, it has to be declared.
We don't have (yet) support for auto backing fields.